apps/range-voting/app/script.js

```diff
@@ -117,98 +118,93 @@ async function loadVoteDescription(vote) {
 
 async function loadVoteData(voteId) {
-  console.info('[RangeVoting > script]: loadVoteData')
+  console.info('[RangeVoting > script]: loadVoteData', voteId)
   let vote
   return new Promise((resolve, reject) => {
     app
       .call('getVote', voteId)
-      .first()
-      .subscribe(voteData => {
+      .pipe(first())
+      .subscribe(async voteData => {
         let funcSig = voteData.executionScript.slice(58, 66)
         if (funcSig == 'f2122136') {
           console.log('Loading Projects Data')
           resolve(loadVoteDataProjects(voteData, voteId))
         } else {
           console.log('Loading Allocations Data')
+          console.log('vote data: ', await loadVoteDataAllocation(voteData, voteId))
           resolve(loadVoteDataAllocation(voteData, voteId))
         }
       })
   })
 }
+
 // These functions arn't DRY make them better
-async function loadVoteDataAllocation(vote, voteId) {
-  return new Promise(resolve =>
+function loadVoteDataAllocation(vote, voteId) {
+  return new Promise(resolve => {
     combineLatest(
       app.call('getVoteMetadata', voteId),
       app.call('getCandidateLength', voteId),
       app.call('canExecute', voteId)
     )
-      .first()
-      .subscribe(([ metadata, totalCandidates, canExecute, payout ]) => {
-        loadVoteDescription(vote).then(async vote => {
-          let options = []
-          for (let i = 0; i < totalCandidates; i++) {
-            let candidateData = await getAllocationCandidate(voteId, i)
-            console.log(candidateData)
-            options.push(candidateData)
-          }
-          let returnObject = {
-            ...marshallVote(vote),
-            metadata,
-            canExecute,
-            options: options,
-          }
-          allocations
-            .getPayout(vote.externalId)
-            .first()
-            .subscribe(payout => {
-              resolve({
-                ...returnObject,
-                limit: parseInt(payout.limit, 10),
-                balance: parseInt(vote.executionScript.slice(706, 770), 16),
-                metadata:
-                  'Range Vote ' +
-                  voteId +
-                  ' - Allocation (' +
-                  payout.metadata +
-                  ')',
-              })
-            })
-        })
+      .pipe(first())
+      .subscribe(async ([ metadata, totalCandidates, canExecute, payout ]) => {
+        const voteDescription = await loadVoteDescription(vote)
+        let options = []
+        for (let i = 0; i < totalCandidates; i++) {
+          const candidateData = await getAllocationCandidate(voteId, i)
+          options.push(candidateData)
+        }
+        options = await Promise.all(options)
+
+        const returnObject = {
+          ...marshallVote(voteDescription),
+          metadata,
+          canExecute,
+          options,
+        }
+
+        allocations.getPayout(voteDescription.externalId)
+          .pipe(first())
+          .subscribe(payout => (resolve({
```

Comment:
This subscription will potentially have some grim merge conflicts, no changes needed here but something to watch for.